PostHog/PostHogSDK.swift

```diff
@@ -331,10 +335,10 @@ let maxRetryDelay = 30.0
         queue.add(PostHogEvent(
             event: event,
             distinctId: getDistinctId(),
-            properties: buildProperties(properties: properties,
-                                        userProperties: userProperties,
-                                        userPropertiesSetOnce: userPropertiesSetOnce,
-                                        groupProperties: groupProperties)
+            properties: buildProperties(properties: sanitizeDicionary(properties),
+                                        userProperties: sanitizeDicionary(userProperties),
+                                        userPropertiesSetOnce: sanitizeDicionary(userPropertiesSetOnce),
+                                        groupProperties: sanitizeDicionary(groupProperties))
```

Comment:
Could this be done in PostHogEvent instead? Feels like something that could easily get forgotten about, not to mention the code duplication

Reply:
The reason for not doing this on a top-level function is that it is an all-in or nothing, in this case, if there's something messed up in `groupProperties`, just `groupProperties` will be dropped but everything else is fine.